{{ cookiecutter.formal_name }}/Dockerfile

```diff
@@ -11,43 +11,45 @@ ARG PY_VERSION
 ARG SYSTEM_REQUIRES
 
 # Make sure installation of tzdata is non-interactive
-ENV DEBIAN_FRONTEND="noninteractive"
+ENV DEBIAN_FRONTEND="noninteractive" \
+# Disable pip warnings about running as root (pip >= 22.1)
+    PIP_ROOT_USER_ACTION="ignore" \
+# Disable pip progress bar
+    PIP_PROGRESS_BAR="off"
 
 # Install the deadsnakes PPA so we can get arbitrary Python versions
 RUN apt-get update -y && \
-    apt-get install -y \
+    apt-get install --no-install-recommends -y \
         software-properties-common \
         dirmngr \
         apt-transport-https \
         lsb-release \
-        ca-certificates
-RUN apt-add-repository ppa:deadsnakes/ppa
-
-# Install git, Python, and any packages required
-RUN apt-get update -y && \
-    apt-get install -y \
+        ca-certificates && \
+    apt-add-repository ppa:deadsnakes/ppa && \
```

Comment:
If this group of packages is specifically related to adding `deadsnakes`, then I think it would be clearer to keep them with `apt-add-repository` as a separate RUN command.

Reply:
This list of packages are some form of copypasta to bootstrap `apt` in a base ubuntu docker image.

Breaking it down though:
- `software-properties-common` installs the `apt-add-repository` command, so it's required.
- `dirmngr` appears to be (or have been) a requirement for `apt-key` which would commonly be used with adding a repository. It isn't strictly necessary here, though.
- `apt-transport-https` was necessary prior to apt 1.5 (released in 2017) when they added native support for https. No longer necessary.
- `lsb-release`....my guess here a lot of times people will build the URL to put in a apt list file using `lsb_release`. It isn't necessary here, though, since we're adding a defined ppa.
- `ca-certificates` seems to have come late to the game but now most packages list it as a dependency as `software-properties-common` does now. Furthermore, a while back, Christian Heimes [asked](https://github.com/deadsnakes/issues/issues/144) `deadsnakes` to make `ca-certificates` a dependency; they added it at least as a recommends but even then Christian pointed out `software-properties-common` required it. Including explicitly is not necessary.

Now that we understand this chesterton fence.....trimmed down to just `software-properties-common`.

Reply:
Confirming this was definitely copypasta from a version of the deadsnakes docs that is *at least* 3 years old, and originally developed for Ubuntu 16.04. If we only need software-properties-common, that's great.